stock_orderpoint_generator/models/product.py

```diff
@@ -26,53 +27,23 @@ def create(self, vals):
             record.auto_orderpoint_template_ids.create_orderpoints(record)
         return record
 
-    @api.multi
     def write(self, vals):
         result = super().write(vals)
         if vals.get("auto_orderpoint_template_ids"):
             orderpoint_templates = self.mapped("auto_orderpoint_template_ids")
             orderpoint_templates.create_orderpoints(self)
         return result
 
-    def _compute_historic_quantities_dict(
-        self, location_id=False, from_date=False, to_date=False
-    ):
-        """Returns a dict of products with a dict of historic moves as for
-           a list of historic stock values resulting from those moves. If
-           a location_id is passed, we can restrict it to such location"""
-        location = location_id and location_id.id
-        domain_quant_loc, domain_move_in_loc, domain_move_out_loc = self.with_context(
-            location=location
-        )._get_domain_locations()
-        if not to_date:
-            to_date = fields.Datetime.now()
-        domain_move_in = domain_move_out = [
-            ("product_id", "in", self.ids),
-            ("state", "=", "done"),
-        ] + domain_move_in_loc
-        domain_move_out = [
-            ("product_id", "in", self.ids),
-            ("state", "=", "done"),
-        ] + domain_move_out_loc
+    def _get_stock_move_domain(self, domain_move=False, from_date=False, to_date=False):
+        domain = [("product_id", "in", self.ids), ("state", "=", "done")] + domain_move
         if from_date:
-            domain_move_in += [("date", ">=", from_date)]
-            domain_move_out += [("date", ">=", from_date)]
-        domain_move_in += [("date", "<=", to_date)]
-        domain_move_out += [("date", "<=", to_date)]
-        move_obj = self.env["stock.move"]
-        # Positive moves
-        moves_in = move_obj.search_read(
-            domain_move_in, ["product_id", "product_qty", "date"], order="date asc"
-        )
-        # We'll convert to negative these quantities to operate with them
-        # to obtain the stock snapshot in every moment
-        moves_out = move_obj.search_read(
-            domain_move_out, ["product_id", "product_qty", "date"], order="date asc"
-        )
-        for move in moves_out:
-            move["product_qty"] *= -1
-        # Merge both results and group them by product id as key
-        moves = moves_in + moves_out
+            domain += [("date", ">=", from_date)]
+        domain += [("date", "<=", to_date)]
+        return domain
+
+    def _set_product_moves_dict(
+        self, moves=False, location=False, from_date=False, to_date=False
```

Comment:
Same suggestion as above here. 

Can you filter by date with said date value set to `False`?

I think `None` is a better default value here, at least for arguments that can't be used with the `False` value (like `moves`), but I don't know if this an Odoo/OCA standard.

However, if this value is not passed and the default is assumed, you will get an error bellow when running the loop for `moves`. You should make them obligatory args, or test if they are set before using them.

Reply:
Everything is related to orderpoint_template model. These functions are called related to some False fields (that it's possible to contain False in some cases).
With these considerations IMO everithing it's correct and False default in functions it's OK.

Reply:
Yes, I understand you can filter in the ORM with some fields set to `False` (like `location`, right?)
However, in this particular case, if you leave `moves` with `False` as default value and never check if is defined on the function call, you might get an error bellow in the loop:
```python
for move in moves:
```
as you might be trying to loop over a `False` value.

If `moves` is always needed, you should remove the default and leave it as an obligatory argument. Otherwise, you must check if it is still `False` before trying to iterate it (or `None` as default as, if this is a Python object and not a Odoo ORM object, it would make more sense to use that as default)

The same can happen with the dates. Does the ORM allow you to filter for `from_date=False` or `to_date=False`? 

> Everything is related to orderpoint_template model. 

I didn't really understand this part. Are you inheriting and overriding this method in particular?